open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +257,427 @@ public static JSONArray appendDummyAccountID() {
         return accountsJSON;
 
     }
+
+    /**
+     * Method that invokes the relevant methods to populate data for each flow.
+     *
+     * @param consentResource Consent Resource parameter containing consent related information retrieved from database
+     * @return ConsentDataJson array
+     */
+    public static JSONArray getConsentData(ConsentResource consentResource) {
+
+        JSONArray consentDataJSON;
+        try {
+            consentDataJSON = new JSONArray();
+            String receiptString = consentResource.getReceipt();
+            Object receiptJSON = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(receiptString);
+
+            // Checking whether the request body is in JSON format
+            if (!(receiptJSON instanceof JSONObject)) {
+                log.error(ErrorConstants.NOT_JSON_OBJECT_ERROR);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, ErrorConstants.NOT_JSON_OBJECT_ERROR);
+            }
+
+            // Checking whether the consent status is valid
+            if (!consentResource.getCurrentStatus().equals(ConsentExtensionConstants.AWAITING_AUTH_STATUS)) {
+                log.error(ErrorConstants.STATE_INVALID_ERROR);
+                // Currently throwing an error as a 400 response.
+                // Developers have the option of appending a field IS_ERROR to the jsonObject
+                // and showing it to the user in the webapp. If so, the IS_ERROR has to be checked in any later steps.
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.STATE_INVALID_ERROR);
+            }
+
+            JSONObject receipt = (JSONObject) receiptJSON;
+
+            // Checks if 'data' object is present in the receipt
+            if (receipt.containsKey(ConsentExtensionConstants.DATA)) {
+                JSONObject data = (JSONObject) receipt.get(ConsentExtensionConstants.DATA);
+
+                String type = consentResource.getConsentType();
+                switch (type) {
+                    case ConsentExtensionConstants.ACCOUNTS:
+                        consentDataJSON = populateAccountData(data, consentDataJSON);
+                        break;
+                    case ConsentExtensionConstants.PAYMENTS:
+                        consentDataJSON = populatePaymentData(data, consentDataJSON);
+                        break;
+                    case ConsentExtensionConstants.FUNDSCONFIRMATIONS:
+                        consentDataJSON = populateCofData(data, consentDataJSON);
+                        break;
+                    case ConsentExtensionConstants.VRP:
+                        // Check if 'controlParameters' object is present in the 'data'
+                        if (data.containsKey(ConsentExtensionConstants.CONTROL_PARAMETERS)) {
+                            JSONObject controlParameters = (JSONObject) data.
```

Comment:
Any reason we fetched `controlParameters ` from the `data` and passed to `populateVRPData()` , for other methods `populatePaymentData(data, consentDataJSON);` , We seems to pass `data` object itself.
For consistency shall we pass `data` object for `populateVRPData` WDYT?

```populateVRPData(data, consentDataJSON);```

Reply:
Fixed in [fa29ae7]